src/libraries/System.Private.CoreLib/src/System/Activator.RuntimeType.cs

```diff
@@ -83,7 +84,7 @@ public static partial class Activator
                                           ref stackMark);
         }
 
-        public static object? CreateInstance(Type type, bool nonPublic) =>
+        public static object? CreateInstance([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.NonPublicConstructors)] Type type, bool nonPublic) =>
```

Comment:
Wouldn't this only need DefaultConstructor?  Or does DefaultConstructor imply only a public parameterless ctor rather than just implying a parameterless ctor?

Reply:
`DefaultConstructor` only means "the constructor that the C# compiler creates for you by default", so it's public+parameterless. If you have a better naming suggestion, maybe it would be worth renaming the enum value. This question keeps popping up....

Reply:
> DefaultConstructor only means "the constructor that the C# compiler creates for you by default", so it's public+parameterless.

Does that mean it'll also keep a protected parameterless ctor on an abstract class?

And presumably it'll keep it even if you wrote it manually rather than the compiler creating it for you.

We need a better name.

Reply:
> We need a better name.

`PublicParameterlessConstructor`?

Reply:
I think that's better.  It also maps to the language used in CS0310 for the new() constraint:
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs0310
https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKBrIAJjyB2Ggbxvq8fKUZXoBZbAEsAdgAoAlPQC8APh4BOCQGFY2DDAA8AMQgR50qQG5O3JnwAq9dTE06rRmQHcAFjFj0bIemJgu0nKK/i7e0mbUAL40NOJaUGLYADaMpPT6EOzmXPGeSamZQWwx0UA=

Reply:
Do I need to file an issue so that it goes through API review, or can we agree on that here and just do it?

Reply:
@terrajobst?

Reply:
Might want to catch him on IM or email.

Reply:
We'll tackle the renaming of DefaultConstructor in a separate pull request. It's a bit out of scope for annotating corelib

Reply:
Yup

Reply:
@stephentoub @MichalStrehovsky email to FXDC is fine for small things like this. FWIW, I think `PublicParameterlessConstructor` is perfect.